### ISSUE TITLE
Screen Reader is able to access UI behind a blocking Modal

### ISSUE THREAD

**Issue by rarattay** [microsoft/fluentui#15125]
### Environment Information

- **Package version(s)**: 7.126.1
- **Browser and OS versions**: iOS  13.7 (mobile) & 13.6.1 (tablet), Android 10 (mobile) & 9 (tablet)

### Describe the issue:

Mobile screen readers are able to escape a blocking or modeless modal dialog. This enables screen reader users to access and interact with some web elements behind the modal, which is incorrect. I have not been able to repro the bug on PC (Edge + narrator) but it does repro on multiple versions of iOS voice over and Andoid talkback. I have also been unable to repro in Code Pen (Possible due to how Code Pen encapsulates the modality into the viewer space or due to lack of UI elements behind the modal. However it does repro on the Fluent UI site example: https://developer.microsoft.com/en-us/fluentui#/controls/web/modal

### Please provide a reproduction of the issue in a codepen:
1. Open https://developer.microsoft.com/en-us/fluentui#/controls/web/modal on an Android or iOS device
2. Enable talkback (Android) or voiceover(iOS)
3. Navigate to & click to open Modeless modal button
4. Navigate through entire modal dialog via swipes
5. Notice that after reaching the end of the modal, the next swipe takes the focus outside of the modal. 

**Can also repro by swiping in the opposite direction through the entire modal. (top to bottom or bottom to top)

#### Actual behavior:

After swiping through the entire modal, the narrator goes outside the modal. This enables users to bypass the modal and interact with other UI elements on the page.

#### Expected behavior:

The screen narrator should not be able to leave the modal dialog.

**Comment by paulgildea** [comment]
@rarattay Thanks for the feedback and providing a preliminary repos. Super helpful.

@smhigley FYI for Talkback/VoiceOver

**Comment by smhigley** [comment]
Sorry if I'm misunderstanding, but isn't the entire point of a "modeless" dialog to allow access to the background page? As long as it's not reproducing for the actual modal dialog, this seems like expected behavior to me.

Side note -- I get why, but the fact that a control is called a "modeless modal" is making me lol pretty hard 😂

**Comment by rarattay** [comment]
According to the documentation on the Fluent UI Modal docs isModeless is: "Whether the dialog should be modeless (e.g. not dismiss when focusing/clicking outside of the dialog). if true: isBlocking is ignored, there will be no overlay (isDarkOverlay is ignored), isClickableOutsideFocusTrap is true, and forceFocusInsideTrap is false".  This leads me to believe that if users are unable to click outside the dialog that screen readers should also not be able to. 

In my specific project we are actually using isBlocking=true rather than isModeless=true, but both seem to cause the same issue and are mostly likely related.

**Comment by smhigley** [comment]
@rarattay modeless dialogs sit in a pretty weird area of interaction, so I understand the confusion :). The relevant bit here is that when a modeless dialog is open, a pointer user can access and interact with the rest of the page without first dismissing the dialog. The dialog may or may not close as a result of that interaction, but it does not *prevent* the interaction by being open. When that is the case, screen reader users should have the same opportunity. Some screen readers do not respect `aria-modal="false"` on `role="dialog"` currently, which is why you might find some inconsistent results and some screen readers that do treat the modeless dialog as modal.

`isBlocking` does something else entirely: it prevents clicking on the overlay to dismiss the dialog, and I assume also prevents the Escape key from dismissing it. `isBlocking` is essentially a way to force a user to take some action within a dialog before they can move on. A modeless dialog cannot be blocking by definition, since it allows access to the rest of the page even when open. If you're encountering a problem with a screen reader accessing the background page when `isBlocking` is true (and `isModeless` is false), then that would be a bug. I can't currently reproduce that behavior on iOS with VoiceOver, but if you can repro with VO or Talkback, versions and a codepen (or similar) would help a lot in debuging 🙂

**Comment by rarattay** [comment]
Do I need to explicitly state that isModeless=false to get the behavior I want (screen reader is unable to leave modal)?

Currently my modal that repros the error looks like:     
<Modal styles={styles} isOpen={isModalOpen} isBlocking={true} allowTouchBodyScroll={false}>

**Comment by smhigley** [comment]
@rarattay I can't reproduce on this codepen: https://cdpn.io/smhigley/debug/b769957a8837ddffa15fab81e4946e68 with VoiceOver on iOS 13.7. Does that one work for you with VoiceOver?

**Comment by JustSlone** [comment]
Confirmed no repro on iOS 13.6 👍 
I did see repro on Android + Talkback however ☹️

**Comment by smhigley** [comment]
I just tested on Chrome + Talkback on Android 10, and could also repro on both this and the [APG modal](https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html). It looks like `aria-modal` isn't supported on Chrome/Android yet.

I think that leaves roughly three options:
1) use `inert` or `aria-hidden="true"` on all the modal's siblings and parent-siblings to hide inactive DOM elements
2) look into expected timelines for google fixing aria-modal support without changing the Fluent dialog
3) recommend teams make their modals top-level in the DOM, and handle putting `aria-hidden="true"` on the (ideally single) inactive container element containing the rest of the page content when the modal opens

A year ago I'd say that 1 or 3 were the only viable options, but support for `aria-modal="true"` has gone up a lot lately. I'll test a bit more to check how many other screen reader combos also fail, but maybe a combination of 2 + 3 could work? Unless you have a good idea for how to implement 1 :)

**Comment by rarattay** [comment]
Is there any update or an ETA on when this will be resolved? I updated my initial description to reflect that it is Android only.
